include/exadg/solvers_and_preconditioners/multigrid/multigrid_preconditioner_base.cpp

```diff
@@ -360,11 +364,153 @@ create_geometric_coarsening_sequence(Triangulation<dim, spacedim> const & fine_t
       // actually create triangulation
       new_tria->create_triangulation(construction_data);
 
+      // save mesh
+      coarse_grid_triangulations[min_level - 1] = new_tria;
+      temp_tria.coarsen_global();
```

Comment:
Doesn't this mean you coarsen once more in the end where it isn't really needed, or do I overlook something?

Reply:
At this spot we actually still want to coarsen (unless the level is the last one, i.e., 1), because we want to further process the coarsened mesh somewhere else. I tried to rearrange the place where we coarsen but think that this place is the most intuitive for the rest of the function where we want to replicate some of the mesh.